backend/src/ml_space_lambda/utils/iam_manager.py

```diff
@@ -435,21 +444,73 @@ def _detach_iam_policies(self, iam_role_name: str) -> List[str]:
 
         return detached_iam_policies
 
-    def _generate_project_policy(self, project: str) -> str:
-        return (
-            self.project_policy.replace("\n", "")
-            .replace("$PROJECT_NAME", project)
-            .replace("$BUCKET_NAME", self.data_bucket)
-            .replace("$PARTITION", self.aws_partition)
-        )
+    """
+    Generates a policy from the provided template and variable replacments
 
-    def _generate_user_policy(self, user: str) -> str:
-        return (
-            self.user_policy.replace("\n", "")
-            .replace("$USER_NAME", user)
-            .replace("$BUCKET_NAME", self.data_bucket)
-            .replace("$PARTITION", self.aws_partition)
-        )
+    Inputs:
+    policy_template: str - A template that includes variables of the form $VAR_NAME
+    replacments: list[list[str]] - A list of str tuples that are to be replaced of the form ["$VAR_NAME", "var value"]
+    """
+
+    def generate_policy_from_template_string(self, policy_template: str, replacements: list[list[str]]):
```

Comment:
Do all of these need to be called externally? If not I'd recommend adding the `_` prefix back.

Reply:
This was a refactor that was done when I thought I would use templates. This could be reverted if we want, but I will keep it and add the `_` for now